content/en/blog/_posts/2024-08-19-pod-failure-policy-for-jobs-goes-ga.md

```diff
@@ -0,0 +1,184 @@
+failures and continue running to completion.
+
+To allow for these transient failures, Kubernetes Jobs include the `backoffLimit`
+field, which lets you specify a number of Pod failures that you're willing to tolerate
+during Job execution. However, if you set a large value for the `backoffLimit` field
+and rely solely on this field, you might notice unnecessary increases in operating
+costs as Pods restart excessively until the backoffLimit is met.
+
+This becomes particularly problematic when running large-scale Jobs with
+thousands of long-running Pods across thousands of nodes.
+
+The Pod failure policy extends the backoff limit mechanism to help you reduce
+costs in the following ways:
+
+- Gives you control to fail the Job as soon as a non-retriable Pod failure occurs.
+- Allows you to ignore retriable errors without increasing the `backoffLimit` field.
+
+For example, you can use a Pod failure policy to run your workload on more affordable spot machines
+by ignoring Pod failures caused by
+[graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown).
+
+The policy allows you to distinguish between retriable and non-retriable Pod
+failures based on container exit codes or Pod conditions in a failed Pod.
+
+## How it works
+
+You specify a Pod failure policy in the Job specification, represented as a list
+of rules.
+
+For each rule you define _match requirements_ based on one of the following properties:
+
+- Container exit codes: the `onExitCodes` property.
+- Pod conditions: the `onPodConditions` property.
+
+Additionally, for each rule, you specify one of the following actions to take
+when a Pod matches the rule:
+- `Ignore`: Do not count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+- `FailJob`: Fail the entire Job and terminate all running Pods.
+- `FailIndex`: Fail the index corresponding to the failed Pod.
+  This action works with the [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index) feature.
+- `Count`: Count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When Pod failures occur in a running Job, Kubernetes matches the
+failed Pod status against the list of Pod failure policy rules, in the specified
+order, and takes the corresponding actions for the first matched rule.
+
+Note that when specifying the Pod failure policy, you must also set the Job's
+Pod template with `restartPolicy: Never`. This prevents race conditions between
+the kubelet and the Job controller when counting Pod failures.
+
+### Kubernetes-initiated Pod disruptions
+
+To allow matching Pod failure policy rules against failures caused by
+disruptions initiated by Kubernetes, this feature introduces the `DisruptionTarget`
+Pod condition.
+
+Kubernetes adds this condition to any Pod, regardless of whether it's managed by
+a Job controller, that fails because of a retriable
```

Comment:
Thank you for reviewing and the remark.

However, I would propose slightly different phrasing, because the sentence before the list of conditions feels a bit out of context, and the previous sentence already says "retrieable".

So, I tentatively pushed commit with the following note:
```
Importantly, other disruption scenarios, like eviction due to exceeding
[Pod container limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/),
do not receive the `DisruptionTarget` condition because they are likely caused
by the Pod itself and would reoccur on retry.
```
WDYT @smarterclayton  @shannonxtreme ?

Reply:
You don't need to say "Importantly", but also, this is something that should be in the doc (https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions) but isn't.

Reply:
I have opened the website PR: https://github.com/kubernetes/website/pull/47378, but given that we passed Docs Freeze since 30th July we probably will only be able to merge in the next release cycle.